### DISCUSSION TITLE
Reqable Help

### DISCUSSION THREAD

**Discussion by MegatronKing** [reqable/reqable-app#583] [Q&A]
Ask questions you’re wondering about.

**Comment by CMJNB** [comment]
在api接口设置里开启了“使用cookie”，也确实获取到了cookie并在cookie管理器里看到。
但是新的同站api，并没有自动在请求时带上cookie。是要在请求头里填上“Cookie”么？值为空的话，依然没有携带cookie。
如何自动在同站api发送请求时携带cookie？
![image](https://github.com/reqable/reqable-app/assets/53365071/59b6c8c2-5eb7-4e0d-83de-42cbecd3c710)
如图，勾选了“使用cookie”，但是请求头里的Cookie要自己添加

**Comment by MegatronKing** [comment]
@CMJNB 看起来像是bug，但是我没有复现。你试试请求`https://baidu.com`，会自动携带cookie。

**Reply by CMJNB** [reply]
第一次请求：`https://baidu.com`，无Cookie，返回Cookie
第二次请求：自动携带Cookie，并无法编辑。符合预期
但是请求我的api却无法达到这种效果。

**Reply by CMJNB** [reply]
找到问题了。
如果Cookie的路径不是根`Path=/;`，则“自动携带Cookie不会起作用”。
假设站点是`http://xx.xx/Self/login/`，返回的Cookie路径为`Path=/Self;`，则不会自动携带这个Cookie。

**Reply by MegatronKing** [reply]
我测试没有这个问题，请求路径`/Self/login/`是能匹配上Cookie Path `/Self`的，有没有可能是过期了？

**Reply by CMJNB** [reply]
![IMG_20240521_190004.png](https://github.com/reqable/reqable-app/assets/53365071/3847b87f-f44c-4e50-8ba2-785ac765ac21)

![Screenshot_2024-05-21-18-58-30-815_com.oray.sunlogin-edit.jpg](https://github.com/reqable/reqable-app/assets/53365071/0d33c032-d01f-4172-94bd-9c95cf69c76d)

![Screenshot_2024-05-21-19-01-25-468_com.oray.sunlogin-edit.jpg](https://github.com/reqable/reqable-app/assets/53365071/867b1bb0-9ef6-4b1b-bfbd-71f669879039)



会不会是存入时大写变小写了，然后没匹配上

**Reply by MegatronKing** [reply]
> 会不会是存入时大写变小写了，然后没匹配上

检查了下代码，Set-Cookie的值都直接转成小写了，逻辑在dart内置库里面，我也不明白为什么要这么做，确实是导致你这个问题的原因。

**Reply by CMJNB** [reply]
![Screenshot_2024-05-21-20-19-03-040_com.oray.sunlogin-edit.jpg](https://github.com/reqable/reqable-app/assets/53365071/6a09714e-41ab-4945-8d79-17d2870639f3)



我看了一下浏览器，是不会改变大小写的，这个应该是库的缺陷？毕竟URL是严格大小写的

**Reply by MegatronKing** [reply]
给他们提了个issue：https://github.com/dart-lang/sdk/issues/55795
下个版本，我先暂时把匹配逻辑改成非大小写敏感。

**Reply by 13MVK26** [reply]
"The problem is that after jailbreaking my iPhone XR on iOS 14.3, the Reqable app from the App Store won’t launch—it crashes instantly when I try to open it. I’ve tested this with every version of unc0ver and just tried Taurine, but the issue remains. How can I fix this?"

**Reply by MegatronKing** [reply]
> "The problem is that after jailbreaking my iPhone XR on iOS 14.3, the Reqable app from the App Store won’t launch—it crashes instantly when I try to open it. I’ve tested this with every version of unc0ver and just tried Taurine, but the issue remains. How can I fix this?"

This is a known issue, but I didn't have such a device to test.

**Comment by 13MVK26** [comment]
But I have such a device, maybe I can help you with something?


пн, 7 апр. 2025 г. в 11:24, Megatron King ***@***.***>:
